index.js

```diff
@@ -122,8 +135,8 @@ function Hook (modules, options, onrequire) {
       moduleName = parsedPath.name
       basedir = parsedPath.dir
     } else {
-      const stat = parse(filename)
-      if (stat === undefined) {
+      const stat = moduleDetailsFromPath(filename)
+      if (!stat) {
```

Comment:
What's the reason for converting `stat === undefined` to `!stat`? It's just an optimisation nit, and probably something that the V8 engine can optimise for, so there's no performance difference anyway - I'm not sure 🤔 

```suggestion
      const stat = moduleDetailsFromPath(filename)
      if (stat === undefined) {
```

Reply:
When I was doing recent work on the APM agent, I had made two changes to this module: one was the `Module._resolveFilename`-related change that I've put in this PR, the other is a change to `moduleDetailsFromPath` that I was going to include as a separate PR after this one. A part of that change to `moduleDetailsFromPath` was changing from `undefined` to `null` as the "I couldn't find a value" value.

Anyway, that is all for a separate PR, and doesn't need to be included here. I switch it back to the `=== undefined` check. Thanks.